verkkokauppa/order/types.py

```diff
@@ -0,0 +1,215 @@
+                        quantity=item["quantity"],
+                        unit=item["unit"],
+                        row_price_net=Decimal(item["rowPriceNet"]),
+                        row_price_vat=Decimal(item["rowPriceVat"]),
+                        row_price_total=Decimal(item["rowPriceTotal"]),
+                        price_net=Decimal(item["priceNet"]),
+                        price_gross=Decimal(item["priceGross"]),
+                        price_vat=Decimal(item["priceVat"]),
+                        vat_percentage=Decimal(item["vatPercentage"]),
+                        meta=[
+                            OrderItemMeta(
+                                order_item_meta_id=UUID(meta["orderItemMetaId"]),
+                                order_item_id=UUID(meta["orderItemId"]),
+                                order_id=UUID(meta["orderId"]),
+                                key=meta["key"],
+                                value=meta["value"],
+                                label=meta["label"],
+                                visible_in_checkout=meta["visibleInCheckout"]
+                                not in [False, "false"],
+                                ordinal=meta["ordinal"],
+                            )
+                            for meta in item["meta"]
+                        ],
+                        period_frequency=item.get("periodFrequency", None),
+                        period_unit=item.get("periodUnit", None),
+                        period_count=item.get("periodCount", None),
+                        start_date=cls._parse_datetime(item.get("startDate", None)),
+                        billing_start_date=cls._parse_datetime(
+                            item.get("billingStartDate", None)
+                        ),
+                    )
+                    for item in json["items"]
+                ],
+                price_net=Decimal(json["priceNet"]),
+                price_vat=Decimal(json["priceVat"]),
+                price_total=Decimal(json["priceTotal"]),
+                checkout_url=json["checkoutUrl"],
+                customer=OrderCustomer(
+                    first_name=json["customer"]["firstName"],
+                    last_name=json["customer"]["lastName"],
+                    email=json["customer"]["email"],
+                    phone=json["customer"]["phone"],
+                ),
+                status=json["status"],
+                subscription_id=UUID(json["subscriptionId"])
+                if json["subscriptionId"]
+                else None,
+                type=json["type"],
+            )
+        except (KeyError, ValueError) as e:
+            raise ParseOrderError("Could not parse order") from e
+
+    @classmethod
+    def _parse_datetime(cls, string: Optional[str]) -> Optional[datetime]:
+        if string is None:
+            return None
+        return datetime.strptime(string, "%Y-%m-%dT%H:%M:%S.%f").replace(
+            tzinfo=timezone.utc
```

Comment:
Good point, I checked and the values returned are 2 hours behind current Finnish time, so I assumed they would always be UTC.

However, [looking at the code](https://github.com/City-of-Helsinki/verkkokauppa-core/blob/ebe8691f3484042202f0afd3cd0404ffb1746be0/common/src/main/java/fi/hel/verkkokauppa/common/util/DateTimeUtil.java), it looks like they use local date/time, which means it will depend on the server clock.

I think I will remove the explicit timezone because we can't trust that it's really UTC.

Reply:
Fixed. 🙂